src/pages/RepoPage/BundlesTab/BundleOnboarding/BundleOnboarding.tsx

```diff
@@ -48,6 +57,16 @@ const Content: React.FC = () => {
 }
 
 const BundleOnboarding: React.FC = () => {
+  const { provider, owner, repo } = useParams<URLParams>()
+  const { data } = useRepo({ provider, owner, repo })
+  const { hardRedirect } = useRedirect({ href: `/${provider}` })
+
+  // if no upload token redirect
+  if (!data?.repository?.uploadToken) {
+    hardRedirect()
+    return <NotFound />
```

Comment:
So for this, this is pretty much a direct copy over from the new repo page, as to why it has both, I'm not 100% certain, I assume it may be incase something doesn't happen fast enough it'll render the not found just incase.

Reply:
I see the commit history on there. Looks like there was inconsistent behaviour with the redirect. Prefer adding a comment explaining the possibly redundant component.

Reply:
Done